migration/idempotent/001-base.sql

```diff
@@ -3375,24 +3375,29 @@ GRANT EXECUTE ON FUNCTION _prom_catalog.insert_metric_row(TEXT, TIMESTAMPTZ[], D
 -- Suppresses corresponding DDL logging, otherwise PG log may get unnecessarily verbose.
 -- Temporary table is created using supplied table and schema as prototype.
 -- Returns temporary table name
-CREATE OR REPLACE FUNCTION _prom_catalog.create_ingest_temp_table(table_name TEXT, schema_name TEXT)
+CREATE OR REPLACE FUNCTION _prom_catalog.create_ingest_temp_table(table_name TEXT, schema_name TEXT, table_prefix TEXT)
     RETURNS TEXT
     SECURITY DEFINER
     VOLATILE
     SET search_path = pg_catalog, pg_temp
-AS $func$             
+AS $func$
+DECLARE
+    temp_table TEXT;         
 BEGIN
     SET LOCAL log_statement = 'none';
+    -- api user has to make sure that table_prefix is unique per session
+    -- this is to prevent different truncated table names having same temp table
+    temp_table := CONCAT(table_prefix, table_name);
```

Comment:
Let's explicitly truncate the identifier here. I've chosen 62 characters for the same reason as it has been chosen in `_prom_catalog.pg_name_unique`.
```suggestion
    temp_table := left(CONCAT(table_prefix, table_name), 62);
```